libnmstate/nm/nmclient.py

```diff
@@ -220,3 +234,53 @@ def _register_first_action(self):
     def _execute_action_once(self, _):
         self.execute_next_action()
         return False
+
+
+class _NmClient:
+    def __init__(self):
+        self.client = NM.Client.new(None)
+        logging.debug("NM.Client created")
+        self._mainloop = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, _exc_type, _exc_value, _traceback):
+        pass
+
+    def __del__(self):
+        try:
+            ref_count = sys.getrefcount(self.client)
+            if ref_count != 2:
+                raise error.NmstateInternalError(
+                    "_NmClient.__del__() with self.client holding "
+                    f"unexpected(not 2) reference count {ref_count}"
+                )
+            if NM.Client.get_version(self.client) >= StrictVersion("1.21.0"):
+                self._nmclient_cleanup()
+            else:
+                self._nmclient_cleanup_old()
+        except Exception:
+            pass
+
+    def _nmclient_cleanup_old(self):
+        self.client = None
+
+    def _nmclient_cleanup(self):
+        if self.client:
+            self._mainloop = _MainLoop()
+            context_busy_watcher = self.client.get_context_busy_watcher()
+            self._mainloop.push_action(
+                self._wait_nmclient_unref, context_busy_watcher
+            )
+            self.client = None
+            del context_busy_watcher
```

Comment:
`del` and assigning `None` are equivalent, right? If so, can we be consistent here?

Reply:
If I use `context_busy_watcher = None`, I got complain:
`NEWThe value assigned to local variable 'context_busy_watcher' is never used.`

If I use `del self.client`, the future access to `self.client` will cause `AttributeError`.